test/unit/test_diagram.py

```diff
@@ -45,6 +47,37 @@ def test_job_agg_data():
     assert job_agg.number_of_columns() == 6
 
 
+def test_seq():
+    """
+    function to test the Class JobData inside autosubmit/monitor/diagram.py
+    """
+    arr = diagram._seq(100, 2, 10)
+    length = 0
+    for value in arr:
+        length+=1
+        assert value is not None
+        assert isinstance(value, int)
```

Comment:
Just `assert isinstance(value, int)` is enough. It verifies if the value is not `None`, and also instance of `int`, `$ python3 -c 'print(isinstance(None, int))'`.